docs/code-generation/generate-hcl.md

```diff
@@ -11,43 +11,40 @@ next:
   link: '/code-generation/generate-file'
 ---
 
-# HCL Code Generation
+# Generating HCL Code with Terramate
 
-Terramate supports the generation of arbitrary HCL code referencing
-[Terramate defined data](../data-sharing/index.md).
+## Configuration files and `generate_hcl` blocks
 
-The generated code can then be composed/referenced by any Terraform code
-inside a stack (or any other tool that uses HCL, like [Packer](https://www.packer.io/)).
+To begin generating HCL code with Terramate, developers need to define configuration files that specify the desired code components and their corresponding values. Terramate utilizes the `generate_hcl` block to generate HCL code based on these configurations. Multiple generate_hcl blocks can be defined within a [configuration file](https://terramate.io/docs/cli/configuration/), each responsible for generating a specific code component.
```

Comment:
```suggestion
To begin generating HCL code with Terramate, developers need to define configuration files that specify the desired code components and their corresponding values. Terramate utilizes the `generate_hcl` block to generate HCL code based on these configurations. Multiple `generate_hcl` blocks can be defined within a [configuration file](../configuration/index.md), each responsible for generating a specific code component.
```

Reply:
@Adeniyikayodee please never use direct links but instead let's use the internals of Vitepress to link to other documentation

Reply:
@Adeniyikayodee we should also limit the line length to 120 max ...